### PR TITLE
Bundle clean-up

### PR DESCRIPTION
**Pull request by kgaillot**

### CODE REVIEW

File: pengine/notif.c [hunk 1]
```diff
@@ -103,7 +103,9 @@ expand_node_list(GListPtr list, char **uname, char **metal)
         int existing_len = 0;
         node_t *node = (node_t *) gIter->data;
 
-        CRM_ASSERT(node->details->uname);
+        if (node->details->uname == NULL) {
+            continue;
```
Comment (beekhof) [inline pengine/notif.c]: If there is no uname, then some placeholder such as "NA" or "unknown" needs to be used instead.
This index of nodes between the uname and physical host name needs to match exactly.  _Everything_ depends on that.

File: pengine/graph.c [hunk 2]
```diff
@@ -441,17 +441,16 @@ update_colo_start_chain(action_t *action)
 
     if (is_not_set(action->flags, pe_action_runnable) && safe_str_eq(action->task, RSC_START)) {
         rsc = uber_parent(action->rsc);
+        if (rsc->parent) {
```
Comment (beekhof) [inline pengine/graph.c]: This change makes no sense.
What is it based on?  Observed or theoretical problems?
At the very least the comment - which talks about stops - doesn't match the check for a start action.

File: pengine/notif.c (hunk 1 above)
Comment (kgaillot) [reply pengine/notif.c]: expand_node_list() is only used for notify_available_uname, notify_all_uname and notify_all_hosts -- none of those are matched up with the others iiuc

expand_list() is used for the others

File: pengine/graph.c (hunk 2 above)
Comment (kgaillot) [reply pengine/graph.c]: It's your comment :)

By "uber_parent() stops before the bundle" you meant that uber_parent() doesn't go up the rsc->parent chain all the way to the bundle resource. So you added this block (in the later position) to get the bundle resource when rsc is in a bundle. The problem is that where it was, it was apply even when uber_parent() wasn't called. AFAICT the only purpose of this is to adjust the uber_parent() behavior, so it should only be called when that is called.

**Comment by kgaillot** [comment]
Merging for now, we can make further corrections as needed

File: pengine/notif.c (hunk 1 above)
Comment (beekhof) [reply pengine/notif.c]: Ohhh, it won't appear in notify_all_uname OR notify_all_hosts if there is no uname.  Thats fine.

**Pull request merged**